src/mapDataManipulation/mapDataManipulationUtils.ts

```diff
@@ -1,22 +1,59 @@
-import { RgbMappingArrays, Effects, ColorRange } from './const';
+import { Effects, ColorRange, ImageProperties } from './const';
 
-export function prepareRgbMappingArrays(): RgbMappingArrays {
-  return {
-    red: [...Array(256).keys()],
-    green: [...Array(256).keys()],
-    blue: [...Array(256).keys()],
-  };
+export async function getImageData(originalBlob: Blob): Promise<ImageProperties> {
+  let imgObjectUrl: any;
+  try {
+    const imgCanvas = document.createElement('canvas');
+    const imgCtx = imgCanvas.getContext('2d');
+    imgObjectUrl = window.URL.createObjectURL(originalBlob);
+    const img: any = await new Promise((resolve, reject) => {
+      const img = new Image();
+      img.onload = () => resolve(img);
+      img.onerror = reject;
+      img.src = imgObjectUrl;
+    });
+
+    imgCanvas.width = img.width;
+    imgCanvas.height = img.height;
+    imgCtx.drawImage(img, 0, 0);
+    const imageData = imgCtx.getImageData(0, 0, img.width, img.height).data;
+    imgCanvas.remove();
+
+    return { imageData, imageWidth: img.width, imageHeight: img.height, imageFormat: originalBlob.type };
+  } catch (e) {
+    console.error(e);
+    return { imageData: new Uint8ClampedArray(), imageWidth: 0, imageHeight: 0, imageFormat: '' };
+  } finally {
+    if (imgObjectUrl) {
+      window.URL.revokeObjectURL(imgObjectUrl);
+    }
+  }
 }
 
-export function changeRgbMappingArraysWithFunction(
-  rgbMappingArrays: RgbMappingArrays,
-  transformationFunction: Function,
-): RgbMappingArrays {
-  const newRgbMappingArrays = { ...rgbMappingArrays };
-  newRgbMappingArrays.red = newRgbMappingArrays.red.map(x => transformationFunction(x));
-  newRgbMappingArrays.green = newRgbMappingArrays.green.map(x => transformationFunction(x));
-  newRgbMappingArrays.blue = newRgbMappingArrays.blue.map(x => transformationFunction(x));
-  return newRgbMappingArrays;
+export async function getBlob(imageProperties: ImageProperties): Promise<Blob> {
+  const { imageData, imageWidth, imageHeight, imageFormat } = imageProperties;
+  let imgObjectUrl: any;
+  try {
+    const imgCanvas = document.createElement('canvas');
+    imgCanvas.width = imageWidth;
+    imgCanvas.height = imageHeight;
+    const imgCtx = imgCanvas.getContext('2d');
+    const newImg = new ImageData(imageData, imageWidth, imageHeight);
+    imgCtx.putImageData(newImg, 0, 0);
+    const blob: Blob = await new Promise(resolve => {
+      imgCanvas.toBlob(blob => {
+        resolve(blob);
+      }, imageFormat);
+    });
+    imgCanvas.remove();
+    return blob;
+  } catch (e) {
+    console.error(e);
+  } finally {
+    if (imgObjectUrl) {
+      window.URL.revokeObjectURL(imgObjectUrl);
+    }
+  }
```

Comment:
This is not needed, maybe we should put `imgCanvas.remove()` here?

Reply:
I think this is correct (done the same way in EOB). Url needs to be revoked. I added removing of canvas to hopefully free a bit of memory.

`imgObjectUrl` is declared outside of the `try-catch` so it can be this way.
`imgCanvas` would need to be declared outside as well if we want to do it this way.

Reply:
Sorry if I was not being clear - this is needed and OK in `getImageData` function, but in `getBlob` function we never assign a value to `imgObjectUrl` at all. :)

Reply:
Ah sorry, totally missed that. I understand now what you are saying. Thanks, will remove it here.

Reply:
Should be visible in the `/utils/canvas.ts` (moved it per suggestion below)